monkey/monkey_island/cc/services/reporting/report.py

```diff
@@ -546,11 +545,16 @@ def report_is_outdated(cls) -> bool:
         return bool(cls._agent_repository.get_agents())
 
     @classmethod
-    def get_report(cls):
+    def update_report(cls):
         if cls._agent_repository is None:
             raise RuntimeError("Agent repository does not exists")
 
         if cls.report_is_outdated():
-            cls._report = safe_generate_regular_report()
+            if not cls._generating_report.is_set():
+                cls._generating_report.set()
+                cls._report = cls.generate_report()
+                cls._generating_report.clear()
```

Comment:
As discussed in planning, Let's use a lock and see if there's any issue before trying a more complicated solution.

Reply:
Done. The generation of the report is not taking anything nearly 10 seconds so I think we are good to go!